### PR TITLE
API diff between .NET 8.0 Preview 4 and .NET 8 Preview 5

### PR DESCRIPTION
**Pull request by carlossanlop**
**Known issues**:
- The AsmDiff tool sometimes skips adding `{` or `{}`. Please add a GitHub suggestion to correct it. [arcade#10981](https://github.com/dotnet/arcade/issues/10981)
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect. I have an issue to track this: [arcade#13814](https://github.com/dotnet/arcade/issues/13814) 
  
Repos area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @dreddy-work @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @jeffhandley @karelz @ericstj @stephentoub

Libraries area owners:
 
- System.Buffers: @dotnet/area-system-buffers 
- System.Collections: @dotnet/area-system-collections 
- System.ComponentModel.DataAnnotations: @dotnet/area-system-componentmodel-dataannotations 
- System.ComponentModel: @dotnet/area-system-componentmodel 
- System.Diagnostics.Metric: @tommcdon @noahfalk 
- System.Diagnostics.Process: @dotnet/area-system-diagnostics-process 
- System.Linq: @dotnet/area-system-linq
- @System: @dotnet/area-system-runtime @dotnet/area-system-memory 
- System.Net: @dotnet/ncl 
- System.Reflection.Emit: @dotnet/area-system-reflection-emit
- System.Reflection.Metadata.Ecma335: @dotnet/area-system-reflection-metadata 
- System.Runtime.InteropServices: @dotnet/interop-contrib 
- System.Runtime.Intrinsics: @dotnet/area-system-runtime-intrinsics 
- System.Security: @dotnet/area-system-security 
- System.Text.Json: @dotnet/area-system-text-json 
- System.Text: @dotnet/area-system-text-encoding 
- System.Web: @Rick-Anderson 
- System.Xml: @dotnet/area-system-xml

### CODE REVIEW

File: release-notes/8.0/preview/api-diff/preview5/Microsoft.NETCore.App/8.0-preview5_System.Buffers.Text.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# System.Buffers.Text
+
+``` diff
+ namespace System.Buffers.Text {
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview5/Microsoft.NETCore.App/8.0-preview5_System.Buffers.Text.md:4]: System.Buffers LGTM

CC. @dotnet/area-system-buffers

File: release-notes/8.0/preview/api-diff/preview5/Microsoft.NETCore.App/8.0-preview5_System.Runtime.Intrinsics.Wasm.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# System.Runtime.Intrinsics.Wasm
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview5/Microsoft.NETCore.App/8.0-preview5_System.Runtime.Intrinsics.Wasm.md:1]: System.Runtime.Intrinsics.Wasm LGTM

CC. @radekdoulik

File: release-notes/8.0/preview/api-diff/preview5/Microsoft.NETCore.App/8.0-preview5_System.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# System
```
Comment (tannergooding) [inline release-notes/8.0/preview/api-diff/preview5/Microsoft.NETCore.App/8.0-preview5_System.md:1]: System LGTM.

CC. @dotnet/area-system-runtime

**Review by eiriktsarpalis — APPROVED** [review]
STJ changes LGTM.

File: release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
+# System.Windows.Forms
+
+``` diff
+ namespace System.Windows.Forms {
+     public class AutoCompleteStringCollection : ICollection, IEnumerable, IList {
+-        bool System.Collections.IList.IsFixedSize { get; }
++        bool IList.IsFixedSize { get; }
```
Comment (dreddy-work) [inline release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md:7]: @carlossanlop, this is coming again. i remember you made changes to ignore only namespace optimization changes in the diff.

**Review by dreddy-work — COMMENTED** [review]
Rest of WinForms LGTM.

File: release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md (hunk 4 above)
Comment (carlossanlop) [reply release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md:7]: Could be another manifestation of the same bug. I'll take a look.

**Review by adamsitnik — APPROVED** [review]
System.Diagnostics.Process LGTM

File: release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+-        int IList.Add(object value);
++        int IList.Add(object? value);
+-        bool IList.Contains(object value);
++        bool IList.Contains(object? value);
+-        int IList.IndexOf(object value);
++        int IList.IndexOf(object? value);
+-        void IList.Insert(int index, object value);
++        void IList.Insert(int index, object? value);
+-        void IList.Remove(object value);
++        void IList.Remove(object? value);
```
Comment (carlossanlop) [inline release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md:23]: Seems only the top 3 properties manifest problem you mentioned, @dreddy-work. Here's the suggestion to fix them:

```suggestion
         bool System.Collections.IList.IsFixedSize { get; }
         bool System.Collections.IList.IsReadOnly { get; }
-        object System.Collections.IList.this[int index] { get; set; }
+        object? System.Collections.IList.this[int index] { get; set; }
-        public event CollectionChangeEventHandler CollectionChanged;
+        public event CollectionChangeEventHandler? CollectionChanged;
-        int IList.Add(object value);
+        int IList.Add(object? value);
-        bool IList.Contains(object value);
+        bool IList.Contains(object? value);
-        int IList.IndexOf(object value);
+        int IList.IndexOf(object? value);
-        void IList.Insert(int index, object value);
+        void IList.Insert(int index, object? value);
-        void IList.Remove(object value);
+        void IList.Remove(object? value);
```

Comment (dreddy-work) [reply release-notes/8.0/preview/api-diff/preview5/Microsoft.WindowsDesktop.App/8.0-preview5_System.Windows.Forms.md]: @carlossanlop, you also see some instances at the bottom of the file. And we may want to go with optimized version (i.e `bool System.Collections.IList.IsFixedSize { get; }` -> `bool IList.IsFixedSize { get; }` to make it consistence and avoid future regression in this tool.

**Pull request merged**